packages/expo-updates/ios/EXUpdates/AppLoader/EXUpdatesRemoteAppLoader.m

```diff
@@ -32,9 +35,27 @@ - (void)loadUpdateFromUrl:(NSURL *)url
                     error:(EXUpdatesAppLoaderErrorBlock)error
 {
   self.manifestBlock = manifestBlock;
-  self.successBlock = success;
   self.errorBlock = error;
+
+  UM_WEAKIFY(self)
```

Comment:
good question! This and `UM_STRONGIFY` are macros defined by unimodules-core to make this pattern easier to write. normally this would look something like this:
```
__weak __typeof__(self) weakSelf = self;
    self.block = ^{
        __typeof__(self) strongSelf = weakSelf;
        [strongSelf doSomething];
        [strongSelf doSomethingElse];
    };
```
This is a common pattern in objective-c to avoid retain cycles by using weak references to `self` in blocks that are saved as properties on `self` -- if we don't weakify, the block will retain `self` automatically and since `self` retains the block, neither object will ever get garbage collected. Usually we don't need to worry about memory management in objective-c but this is one of those weird cases that we do.